helpdesk/tests/test_get_email.py

```diff
@@ -20,35 +20,114 @@
-    ''' Test reading emails from a local directory '''
 
-    def setUp(self):
-        self.queue_public = Queue.objects.create(title='Queue 1', slug='QQ', allow_public_submission=True, allow_email_submission=True, email_box_type='local', email_box_local_dir='/var/lib/mail/helpdesk/')
+class GetEmailCommonTests(TestCase):
 
     # tests correct syntax for command line option
     def test_get_email_quiet_option(self):
+        """Test quiet option is properly propagated"""
         with mock.patch('helpdesk.management.commands.get_email.process_email') as mocked_processemail:
             call_command('get_email', quiet=True)
             mocked_processemail.assert_called_with(quiet=True)
             call_command('get_email')
             mocked_processemail.assert_called_with(quiet=False)
 
-    # tests reading emails from a queue and creating tickets
+
+class GetEmailParametricTemplate(object):
+    """TestCase that checks email functionality accross methods and socks configs."""
+
+    def setUp(self):
+
+        self.temp_logdir = mkdtemp()
+        kwargs = {
+            "title": 'Queue 1',
+            "slug": 'QQ',
+            "allow_public_submission": True,
+            "allow_email_submission": True,
+            "email_box_type": self.method,
+            "logging_dir": self.temp_logdir,
+            "logging_type": 'none'}
+
+        if self.method == 'local':
+            kwargs["email_box_local_dir"] = '/var/lib/mail/helpdesk/'
+        else:
+            kwargs["email_box_host"] = unrouted_email_server
+            kwargs["email_box_port"] = unrouted_port
+
+        if self.socks:
+            kwargs["socks_proxy_type"] = self.socks
+            kwargs["socks_proxy_host"] = unrouted_socks_server
+            kwargs["socks_proxy_port"] = unrouted_port
+
+        self.queue_public = Queue.objects.create(**kwargs)
+
+    def tearDown(self):
+
+        rmtree(self.temp_logdir)
+
     def test_read_email(self):
+        """Tests reading emails from a queue and creating tickets."""
         test_email = "To: update.public@example.com\nFrom: comment@example.com\nSubject: Some Comment\n\nThis is the helpdesk comment via email."
-        with mock.patch('helpdesk.management.commands.get_email.listdir') as mocked_listdir, \
-                mock.patch('helpdesk.management.commands.get_email.isfile') as mocked_isfile, \
-                mock.patch('builtins.open' if six.PY3 else '__builtin__.open', mock.mock_open(read_data=test_email)):
-            mocked_isfile.return_value = True
-            mocked_listdir.return_value = ['filename1', 'filename2']
+        test_mail_len = len(test_email)
 
-            call_command('get_email')
+        if self.socks:
+            from socks import ProxyConnectionError
+            with self.assertRaisesRegexp(ProxyConnectionError, '%s:%s' % (unrouted_socks_server, unrouted_port)):
+                call_command('get_email')
+
+            # with mock.patch('socket.socket') as mocked_socket,\
```

Comment:
Good feedback, I will definitely comment it more, submitted the PR too immediately after writing the code to think of it myself, so thanks! 😄